[HUDI-1769]Add download page to the site

garyli1019 opened the pull request:
## What is the purpose of the pull request

Add the download page to the asf site

## Brief change log

Added download.md and add download tab on the navigation bar.

## Verify this pull request

Built the site locally. Tested on chrome and safari.
![image](https://user-images.githubusercontent.com/23007841/115261859-d87f2480-a0e8-11eb-8a77-c6f3ea62c81b.png)

## Committer checklist

 - [ ] Has a corresponding JIRA in PR title & commit
 
 - [ ] Commit message is descriptive of the change
 
 - [ ] CI is green

 - [ ] Necessary doc changes done or have another open PR
       
 - [ ] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA.

nsivabalan commented on docs/_pages/download.md:
```diff
@@ -0,0 +1,36 @@
+* Release Note : ([Release Note for Apache Hudi 0.5.1](/older-releases#release-051-incubating-docs))
+
+## Release 0.5.0-incubating
+* Source Release : [Apache Hudi 0.5.0-incubating Source Release](https://downloads.apache.org/incubator/hudi/0.5.0-incubating/hudi-0.5.0-incubating.src.tgz) ([asc](https://downloads.apache.org/incubator/hudi/0.5.0-incubating/hudi-0.5.0-incubating.src.tgz.asc), [sha512](https://downloads.apache.org/incubator/hudi/0.5.0-incubating/hudi-0.5.0-incubating.src.tgz.sha512))
```
guess this is the right prefix for urls https://downloads.apache.org/hudi/0.5.0-incubating/ for incubating releases. there is no "incubator" before /hudi/

nsivabalan reviewed (changes requested):
fix all incubating links.

nsivabalan reviewed (approved):
Will wait for a day or two to see if @n3nash / @vinothchandar has any reviews, as we are introducing this page for the first time. Will merge it then.

vinothchandar commented:
@garyli1019 the download links are pointing to dist.apache.org tar balls?? 

while we are at it, can we also update release cwiki page with updating this for each release?

garyli1019 commented:
> @garyli1019 the download links are pointing to dist.apache.org tar balls??
> 
> while we are at it, can we also update release cwiki page with updating this for each release?

@vinothchandar fixed. Changed the link to the mirror site like other apache projects. And added the updating `download.md` step to the README instructions.

nsivabalan commented on docs/_pages/download.cn.md:
```diff
@@ -7,29 +7,29 @@ last_modified_at: 2019-12-30T15:59:57-04:00
 
 ## Release 0.8.0
-* Source Release : [Apache Hudi 0.8.0 Source Release](https://downloads.apache.org/hudi/0.8.0/hudi-0.8.0.src.tgz) ([asc](https://downloads.apache.org/hudi/0.8.0/hudi-0.8.0.src.tgz.asc), [sha512](https://downloads.apache.org/hudi/0.8.0/hudi-0.8.0.src.tgz.sha512))
+* Source Release : [Apache Hudi 0.8.0 Source Release](https://www.apache.org/dyn/closer.lua/hudi/0.8.0/hudi-0.8.0.src.tgz) ([asc](https://downloads.apache.org/hudi/0.8.0/hudi-0.8.0.src.tgz.asc), [sha512](https://downloads.apache.org/hudi/0.8.0/hudi-0.8.0.src.tgz.sha512))
```
https://www.apache.org/dyn/closer.lua/hudi/ redirects me to https://apache.osuosl.org/hudi/ fyi.

nsivabalan replied:
probably it chooses a mirror location closer to your geo location.

garyli1019 replied:
probably. www.apache.org/dyn/closer.lua was also mentioned in the instruction email sent by the owner of announce@apache.org, so I think this should be the right one to put on the site.

vinothchandar commented:
lgtm. @nsivabalan if you are happy too, we can land this
